chain-crypto/src/algorithms/vrf/vrf.rs

```diff
@@ -176,30 +160,27 @@ impl PublicKey {
 }
 
 impl ProvenOutputSeed {
-    pub const BYTES_LEN: usize = dleq::Proof::PROOF_SIZE + GroupElement::BYTES_LEN;
+    pub const BYTES_LEN: usize = dleq::Zkp::BYTES_LEN + GroupElement::BYTES_LEN;
     /// Verify a proof for a given public key and a data slice
     pub fn verify(&self, public_key: &PublicKey, input: &[u8]) -> bool {
-        let dleq = dleq::Dleq {
-            g1: &GroupElement::generator(),
-            h1: &public_key.0,
-            g2: &GroupElement::from_hash(&input),
-            h2: &self.u.0,
-        };
-        dleq::verify(&dleq, &self.dleq_proof)
+        self.dleq_proof.verify(
+            &GroupElement::generator(),
+            &GroupElement::from_hash(&input),
+            &public_key.0,
+            &self.u.0,
+        )
     }
 
     pub fn to_buffer(&self, output: &mut [u8]) {
         assert_eq!(output.len(), Self::BYTES_LEN);
         output[0..GroupElement::BYTES_LEN].copy_from_slice(&self.u.0.to_bytes());
         self.dleq_proof
-            .to_bytes(&mut output[GroupElement::BYTES_LEN..]);
+            .to_mut_slice(&mut output[GroupElement::BYTES_LEN..]);
```

Comment:
I don't like the `to_*` names, both before and after: `to_*` is conventionally used for conversion methods that return a value (most `to_bytes` methods elsewhere conform as they return a fixed-size array).

Can we use the opportunity to start renaming these slice-filling methods to something like `write_to_bytes`?